### ISSUE TITLE
PseudoAnosovPerronRootQ prints warning

### ISSUE THREAD

**Issue by jeanluct** [jeanluct/PseudoAnosov#6]
```
PseudoAnosovPerronRootQ[x^2 - 3 x + 1]

Abs::argx: Abs called with 0 arguments; 1 argument is expected. >>

True
```

Here the optional argument lmax should default to zero, but it doesn't.  The argument list of the functions has `lmax___:0` (three underscores).  Why?

**Comment by jeanluct** [comment]
_From Jean-Luc Thiffeault on 2015-10-02 14:50:30+00:00_

Fix argument pattern macthing for PseudoAnosovPerronRootQ (0 not assigned).
Resolves issue #6.

→ <<cset ac2d7f8cf8de>>

**Comment by jeanluct** [comment]
_From Jean-Luc Thiffeault on 2015-10-02 14:55:50+00:00_

This is an annoying behavior of Mathematica pattern matching.  Need three underscored so that

```
PseudoAnosovPerronRootQ[x^2 - 3 x + 1, EqualityTolerance->10^-9]
```

and

```
PseudoAnosovPerronRootQ[x^2 - 3 x + 1, 3, EqualityTolerance->10^-9]
```

both work.  Here we are skipping the option `lmax`, so `lmax` gets set to empty.  The previous notation, `lmax___:0` doesn't work, since three underscores (`BlankNullSequence`) _always_ matches.

The solution is to replace `lmax___:0` by `lmax___`, and instead to test for `{lmax} =!= {}`.  So `0` is now a legal value for `lmax`, which is fine (nothing will match).  Pretty sure an explicit `0` was used nowhere.
